src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java

```diff
@@ -858,13 +864,13 @@ protected void failoverWriter() throws SQLException {
       this.failoverStartTimeMs = 0;
     }
 
-    if (failoverResult == null || !failoverResult.isConnected()) {
+    if (!failoverResult.isConnected()) {
```

Comment:
I'd keep previous code. It's safer.

Reply:
both the failover procedures don't return null anymore; they always return a failed or successful result. But if you want me to add the check back in case we change the implementation in the future then I can add it back